ssl/internal.h

```diff
@@ -1064,12 +1064,25 @@ enum ssl_private_key_result_t ssl_private_key_sign(
     SSL_HANDSHAKE *hs, uint8_t *out, size_t *out_len, size_t max_out,
     uint16_t sigalg, Span<const uint8_t> in);
 
+// ssl_private_key_sign_pha works the same as |ssl_private_key_sign| but
+// is used in the PHA case where the handshake object is not available. Hints
+// are not supported.
+enum ssl_private_key_result_t ssl_private_key_sign_pha(
+    SSL *ssl, uint8_t *out, size_t *out_len, size_t max_out,
+    uint16_t sigalg, Span<const uint8_t> in);
+
 enum ssl_private_key_result_t ssl_private_key_decrypt(SSL_HANDSHAKE *hs,
                                                       uint8_t *out,
                                                       size_t *out_len,
                                                       size_t max_out,
                                                       Span<const uint8_t> in);
 
+// ssl_private_key_supports_signature_algorithm_pha returns whether |ssl|'s
+// private key supports |sigalg|. Used in the PHA case where hs object is
+// unavailable.
+bool ssl_private_key_supports_signature_algorithm_pha(SSL *ssl,
+                                                      uint16_t sigalg);
+
 // ssl_private_key_supports_signature_algorithm returns whether |hs|'s private
 // key supports |sigalg|.
 bool ssl_private_key_supports_signature_algorithm(SSL_HANDSHAKE *hs,
```

Comment:
`ssl_private_key_supports_signature_algorithm_pha` and `ssl_private_key_supports_signature_algorithm` are only used in one location each: in `tls13_choose_signature_algorithm_pha` and in `tls1_choose_signature_algorithm`, respectively.

Perhaps the signature of `ssl_private_key_supports_signature_algorithm_helper` (with possibly a different name) would be more useful to expose from `ssl_privkey.c`?